rustworkx-core/src/token_swapper.rs

```diff
@@ -0,0 +1,556 @@
+                todo_nodes.push(*node);
+            }
+        }
+        // Add initial edges to the digraph/sub_digraph
+        for node in self.graph.node_identifiers() {
+            self.add_token_edges(
+                self.node_map[&node],
+                &mut digraph,
+                &mut sub_digraph,
+                &mut tokens,
+            );
+        }
+        // Do the trials and build a results Vec
+        let mut trial_results: Vec<Vec<Swap>> = Vec::new();
+        for _ in 0..self.trials {
+            let results = self.trial_map(
+                &mut digraph.clone(),
+                &mut sub_digraph.clone(),
+                &mut tokens.clone(),
+                &mut todo_nodes.clone(),
+                &mut rng_seed,
+            );
+            trial_results.push(results);
+        }
+        // Build the first results Vec until a 0 len result is found
+        let mut first_results: Vec<Vec<Swap>> = Vec::new();
+        for result in trial_results {
+            if result.is_empty() {
+                first_results.push(result);
+                break;
+            }
+            first_results.push(result);
+        }
+        // Return the min of the first results
+        let mut res_min = MAX;
+        let mut final_result: Vec<Swap> = Vec::new();
+        for res in first_results {
+            let res_len = res.len();
+            if res_len < res_min {
+                final_result = res;
+                res_min = res_len;
+            }
+        }
+        final_result
+    }
+
+    fn add_token_edges(
+        &mut self,
+        node: NodeIndex,
+        digraph: &mut StableGraph<(), (), Directed>,
+        sub_digraph: &mut StableGraph<(), (), Directed>,
+        tokens: &mut HashMap<NodeIndex, NodeIndex>,
+    ) {
+        // Adds an edge to digraph if distance from the token to a neighbor is
+        // less than distance from token to node. sub_digraph is same except
+        // for self-edges.
+        if !(tokens.contains_key(&node)) {
+            return;
+        }
+        if tokens[&node] == node {
+            digraph.update_edge(node, node, ());
+            return;
+        }
+        let id_node = self.rev_node_map[&node];
+        let id_token = self.rev_node_map[&tokens[&node]];
+        for id_neighbor in self.graph.neighbors(id_node) {
+            let neighbor = self.node_map[&id_neighbor];
+            let dist_neighbor: Result<DictMap<G::NodeId, usize>, &str> = dijkstra(
+                &self.graph,
+                id_neighbor,
+                Some(id_token),
+                |_| Ok::<usize, &str>(1),
+                None,
+            );
+            let dist_node: Result<DictMap<G::NodeId, usize>, &str> = dijkstra(
+                &self.graph,
+                id_node,
+                Some(id_token),
+                |_| Ok::<usize, &str>(1),
+                None,
+            );
+            if dist_neighbor.unwrap()[&id_token] < dist_node.unwrap()[&id_token] {
```

Comment:
I would move these unwraps to the variable definitions and remove the `Result<>` from the types on `dist_node` and `dist_neighbor`. I just think that would be easier to read since we're never going to error in `dijkstra()` as our `edge_cost` argument is hard coded to 1.

Reply:
Done in https://github.com/Qiskit/rustworkx/pull/765/commits/2d74c45a8e8c6e763cd7ce32034d9f094f991661